product-service/handlers/getProduct.js

```diff
@@ -0,0 +1,24 @@
+import { getMockData } from "../helpers.js"
+
+const getProduct = async (event) => {
+    const { productId } = event.pathParameters
+
+    const data = await getMockData()
+    const productData = data.find((d) => d.id === productId)
```

Comment:
I think productId will be string, but mock contains number as `id`. And for that reason the `products/{productId}` endpoint is not working.